cpp/include/cudf/utilities/span.hpp

```diff
@@ -365,7 +369,7 @@ class base_2dspan {
    * @param data Pointer to the data
    * @param size Size of the 2D span as pair
    */
-  base_2dspan(T* data, size_type size) noexcept : _data{data}, _size{size} {}
+  base_2dspan(T* data, size_type size) noexcept : _data{data}, _size{std::move(size)} {}
```

Comment:
We don't need to use move semantics on a primitive (int) type.

Reply:
Checked the rest with
```
(rapids) coder _ ~/cudf $ git diff upstream/branch-24.08 | grep -E 'std::move' | grep -E '^[+-]'
+      _source{std::move(user_defined_aggregator)},
+    : _flattened_columns{std::move(flattened_columns_)},
+  column_metadata(std::string _name) : name(std::move(_name)) {}
+    : arrow_file(std::move(file))
+    : _sink(std::move(sink)), _table(table), _rows_per_chunk(table.num_rows())
+    : _sink(std::move(sink)), _table(std::move(table)), _rows_per_chunk(table.num_rows())
+    : _sink(std::move(sink)), _table(std::move(table))
+  chunked_orc_writer_options(sink_info sink) : _sink(std::move(sink)) {}
+    : _sink(std::move(sink)), _table(std::move(table))
+  explicit chunked_parquet_writer_options(sink_info sink) : _sink(std::move(sink)) {}
+    : name(std::move(_name)), is_nullable(_is_nullable), is_binary(_is_binary)
+    : schema_idx(_schema_idx), name(std::move(_name)), has_repetition(_has_repetition)
+    : arrow_buffer(std::move(arrow_buffer))
+  : ptx(std::move(ptx_)),
+    function_name(std::move(function_name_)),
+    output_arg_type(std::move(output_arg_type_)),
+    : col(std::move(col)), output_dtype(output_dtype), init(init), mr(mr), stream(stream)
-    src.push_back(cudf::test::structs_column_wrapper(std::move(inputs)));
+    src.emplace_back(std::move(inputs));
-      inner_structs.push_back(cudf::test::structs_column_wrapper(std::move(children)));
+      inner_structs.emplace_back(std::move(children));
-    src.push_back(cudf::test::structs_column_wrapper(std::move(inputs)));
+    src.emplace_back(std::move(inputs));
-      inner_structs.push_back(cudf::test::structs_column_wrapper(std::move(children)));
+      inner_structs.emplace_back(std::move(children));
-    return structs{std::move(expected_vector_of_columns), {0, 1, 1, 1, 1}};
+    return structs{std::move(expected_vector_of_columns), {false, true, true, true, true}};
-    cudf::test::structs_column_wrapper{{std::move(struct_children)}, {0, 0, 1, 1, 1, 1}};
+  auto outer_struct = cudf::test::structs_column_wrapper{{std::move(struct_children)},
+    : gby_column(std::move(gby)),
+      oby_column(std::move(oby)),
+      agg_column(std::move(agg)),
+      preceding(std::move(preceding_scalar)),
+      following(std::move(following_scalar)),
-  cudf::test::structs_column_wrapper valid_struct_col(std::move(src_columns), {1});
+  cudf::test::structs_column_wrapper valid_struct_col(std::move(src_columns), {true});
-  cudf::test::structs_column_wrapper invalid_struct_col(std::move(src_columns), {0});
+  cudf::test::structs_column_wrapper invalid_struct_col(std::move(src_columns), {false});
```
I don't see any other moves that are similarly superfluous.

Reply:
I don't know how to tell clang-tidy that these moves aren't necessary. I can add a NOLINT comment to this line, but [the rule doesn't appear to be customizable in any way](https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html). Let me know what you'd like to do here.

Reply:
I am happy with the analysis you did. I'd accept this as-is and move on. You could add NOLINT if you want, either way is fine.